File: leetcode/0070-climbing-stairs/solution.cpp
class Solution {
public:
    unordered_map<int, int> climbed;

    int climbStairs(int n) {
        if (n == 0 || n == 1) {
            return 1;
        }
        if (climbed.find(n) == climbed.end()) {
            climbed[n] = climbStairs(n - 1) + climbStairs(n - 2);
        }
        return climbed[n];
    }
};

File: leetcode/0070-climbing-stairs/solution.py
class Solution(object):
    climbed = {}

    def climbStairs(self, n):
        if n == 0 or n == 1:
            return 1
        if n not in self.climbed:
            self.climbed[n] = self.climbStairs(n - 1) + self.climbStairs(n - 2)
        return self.climbed[n]